### PR TITLE
Fixes #4260 - Item with P582 (end time) shown as existing

### LINKED ISSUE #4260: Item with P582 (end time) shown as existing
Item https://www.wikidata.org/wiki/Q17183050 (Beer Lane) has a P582 (end time) property:
Screen Shot 2021-02-21 at 12 55 23
Despite that, it is shown by the app as existing:
adb

Latest master

Its pin should be black, and it should not be shown when the "Existing" checkbox is checked. Example of how it should look ([item](https://www.wikidata.org/wiki/Q11321073)):
adb

### PR DESCRIPTION
**Pull request by Pratham2305**
**Description (required)**

Fixes #4260 

What changes did you make and why?

Replaced the content of nearby_query
Added `endtime` in NearByResultItem and Place.java, and also created a boolean `exists` that will tell whether the place exists or not.
Replaced `destroyed` with `exists` in NearByParentFragment and NearByController and few other changes in BookmarkLocationDeo.java and Place.java

**Tests performed (required)**

Tested betaDebug on Pixel 3 with API level 29.

**Screenshot**

Screenshot_1615575356

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/nearby/Place.java [hunk 1]
```diff
@@ -26,12 +26,14 @@
     private final String category;
     public final String pic;
     public final String destroyed;
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/nearby/Place.java:28]: Regarding to this comment of @nicolas-raoul , which I agree https://github.com/commons-app/apps-android-commons/issues/4260#issuecomment-783781535  I think we should remove destroyed and end time properties from Place.java and keep only exists variable instead.

File: app/src/main/java/fr/free/nrw/commons/nearby/Place.java [hunk 2]
```diff
@@ -54,6 +58,9 @@ public Place(Parcel in) {
         this.pic = (picString == null) ? "":picString;
         String destroyedString = in.readString();
         this.destroyed = (destroyedString == null) ? "":destroyedString;
+        String endTimeString = in.readString();
+        this.endTime = (endTimeString == null) ? "":endTimeString;
+        this.exists = (destroyedString == null) && (endTimeString == null);
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/nearby/Place.java:63]: please add a comment next to this && operation where you briefly explained what is happening

File: app/src/main/java/fr/free/nrw/commons/nearby/Place.java [hunk 3]
```diff
@@ -26,12 +26,14 @@
     private final String category;
     public final String pic;
     public final String destroyed;
+    public final String endTime;
+    public final Boolean exists;
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/nearby/Place.java:30]: Please explain what will this exists will hold with a comment. do not forget to mention that some places may have destroyed and some may have end time. It is not guaranteed that we have both of them on the same place even if the place does not exist.

**Review by neslihanturan — CHANGES_REQUESTED** [review]
Thanks for the PR!:) Please see my reviews and tests are broken :)

File: app/src/main/java/fr/free/nrw/commons/nearby/Place.java (hunk 1 above)
Comment (Pratham2305) [reply app/src/main/java/fr/free/nrw/commons/nearby/Place.java]: Sorry about that, I have made the changes now.

File: app/src/main/java/fr/free/nrw/commons/nearby/Place.java (hunk 3 above)
Comment (Pratham2305) [reply app/src/main/java/fr/free/nrw/commons/nearby/Place.java:30]: Done!

File: app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java [hunk 4]
```diff
@@ -1317,7 +1317,7 @@ public void updateMarker(final boolean isBookmarked, final Place place, @Nullabl
             return (isBookmarked ?
                 R.drawable.ic_custom_map_marker_green_bookmarked :
                 R.drawable.ic_custom_map_marker_green);
-        } else if (!place.destroyed.trim().isEmpty()) { // Means place is destroyed
+        } else if (!place.exists) { // Means place is destroyed
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java:1320]: Please reword the comment to `// Means that the topic of the Wikidata item does not exist in the real world anymore, for instance it is a past event, or a place that was destroyed.`
Thanks!

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Comment by nicolas-raoul** [comment]
I am getting this error when running this PR:
```
03-16 00:17:02.915 21430 21430 E SQLiteLog: (1) no such column: location_exists
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: android.database.sqlite.SQLiteException: no such column: location_exists (code 1 SQLITE_ERROR[1]): , while compiling: SELECT location_name, location_language, location_description, location_category, location_label_text, location_label_icon, location_lat, location_long, location_image_url, location_wikipedia_link, location_wikidata_link, location_commons_link, location_pic, location_exists FROM bookmarksLocations
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1229)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:703)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:59)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1865)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:419)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:296)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsContentProvider.query(BookmarkLocationsContentProvider.java:61)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider.query(ContentProvider.java:1078)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider.query(ContentProvider.java:1170)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider$Transport.query(ContentProvider.java:241)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:163)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:146)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:136)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsDao.getAllBookmarksLocations(BookmarkLocationsDao.java:48)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.presenter.NearbyParentFragmentPresenter.updateMapMarkers(NearbyParentFragmentPresenter.java:219)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.NearbyParentFragment.updateMapMarkers(NearbyParentFragment.java:888)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.NearbyParentFragment.lambda$populatePlacesForCurrentLocation$15$NearbyParentFragment(NearbyParentFragment.java:852)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.-$$Lambda$NearbyParentFragment$FKmIMzNuulD6GltQ6RMQzwfMXtg.accept(Unknown Source:6)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:119)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Handler.handleCallback(Handler.java:873)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Handler.dispatchMessage(Handler.java:99)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Looper.loop(Looper.java:214)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.app.ActivityThread.main(ActivityThread.java:7032)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at java.lang.reflect.Method.invoke(Native Method)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
```
Maybe database migration code is needed?

**Comment by Pratham2305** [comment]
> I am getting this error when running this PR:
> 
> ```
> 03-16 00:17:02.915 21430 21430 E SQLiteLog: (1) no such column: location_exists
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: android.database.sqlite.SQLiteException: no such column: location_exists (code 1 SQLITE_ERROR[1]): , while compiling: SELECT location_name, location_language, location_description, location_category, location_label_text, location_label_icon, location_lat, location_long, location_image_url, location_wikipedia_link, location_wikidata_link, location_commons_link, location_pic, location_exists FROM bookmarksLocations
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1229)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:703)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:59)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1865)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:419)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:296)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsContentProvider.query(BookmarkLocationsContentProvider.java:61)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider.query(ContentProvider.java:1078)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider.query(ContentProvider.java:1170)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider$Transport.query(ContentProvider.java:241)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:163)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:146)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:136)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsDao.getAllBookmarksLocations(BookmarkLocationsDao.java:48)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.presenter.NearbyParentFragmentPresenter.updateMapMarkers(NearbyParentFragmentPresenter.java:219)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.NearbyParentFragment.updateMapMarkers(NearbyParentFragment.java:888)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.NearbyParentFragment.lambda$populatePlacesForCurrentLocation$15$NearbyParentFragment(NearbyParentFragment.java:852)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.-$$Lambda$NearbyParentFragment$FKmIMzNuulD6GltQ6RMQzwfMXtg.accept(Unknown Source:6)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:119)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Handler.handleCallback(Handler.java:873)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Handler.dispatchMessage(Handler.java:99)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Looper.loop(Looper.java:214)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.app.ActivityThread.main(ActivityThread.java:7032)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at java.lang.reflect.Method.invoke(Native Method)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
> ```
> 
> Maybe database migration code is needed?

Try to login again, then it will work fine

> I am getting this error when running this PR:
> 
> ```
> 03-16 00:17:02.915 21430 21430 E SQLiteLog: (1) no such column: location_exists
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: android.database.sqlite.SQLiteException: no such column: location_exists (code 1 SQLITE_ERROR[1]): , while compiling: SELECT location_name, location_language, location_description, location_category, location_label_text, location_label_icon, location_lat, location_long, location_image_url, location_wikipedia_link, location_wikidata_link, location_commons_link, location_pic, location_exists FROM bookmarksLocations
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1229)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:703)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:59)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1865)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:419)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:296)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsContentProvider.query(BookmarkLocationsContentProvider.java:61)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider.query(ContentProvider.java:1078)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider.query(ContentProvider.java:1170)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider$Transport.query(ContentProvider.java:241)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:163)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:146)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:136)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsDao.getAllBookmarksLocations(BookmarkLocationsDao.java:48)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.presenter.NearbyParentFragmentPresenter.updateMapMarkers(NearbyParentFragmentPresenter.java:219)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.NearbyParentFragment.updateMapMarkers(NearbyParentFragment.java:888)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.NearbyParentFragment.lambda$populatePlacesForCurrentLocation$15$NearbyParentFragment(NearbyParentFragment.java:852)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.-$$Lambda$NearbyParentFragment$FKmIMzNuulD6GltQ6RMQzwfMXtg.accept(Unknown Source:6)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:119)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Handler.handleCallback(Handler.java:873)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Handler.dispatchMessage(Handler.java:99)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Looper.loop(Looper.java:214)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.app.ActivityThread.main(ActivityThread.java:7032)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at java.lang.reflect.Method.invoke(Native Method)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
> ```
> 
> Maybe database migration code is needed?

Try to login again, then it will work fine

**Comment by Pratham2305** [comment]
> I am getting this error when running this PR:
> 
> ```
> 03-16 00:17:02.915 21430 21430 E SQLiteLog: (1) no such column: location_exists
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: android.database.sqlite.SQLiteException: no such column: location_exists (code 1 SQLITE_ERROR[1]): , while compiling: SELECT location_name, location_language, location_description, location_category, location_label_text, location_label_icon, location_lat, location_long, location_image_url, location_wikipedia_link, location_wikidata_link, location_commons_link, location_pic, location_exists FROM bookmarksLocations
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1229)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:703)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:59)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1865)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:419)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:296)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsContentProvider.query(BookmarkLocationsContentProvider.java:61)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider.query(ContentProvider.java:1078)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider.query(ContentProvider.java:1170)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProvider$Transport.query(ContentProvider.java:241)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:163)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:146)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.content.ContentProviderClient.query(ContentProviderClient.java:136)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsDao.getAllBookmarksLocations(BookmarkLocationsDao.java:48)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.presenter.NearbyParentFragmentPresenter.updateMapMarkers(NearbyParentFragmentPresenter.java:219)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.NearbyParentFragment.updateMapMarkers(NearbyParentFragment.java:888)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.NearbyParentFragment.lambda$populatePlacesForCurrentLocation$15$NearbyParentFragment(NearbyParentFragment.java:852)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at fr.free.nrw.commons.nearby.fragments.-$$Lambda$NearbyParentFragment$FKmIMzNuulD6GltQ6RMQzwfMXtg.accept(Unknown Source:6)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:119)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Handler.handleCallback(Handler.java:873)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Handler.dispatchMessage(Handler.java:99)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.os.Looper.loop(Looper.java:214)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at android.app.ActivityThread.main(ActivityThread.java:7032)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at java.lang.reflect.Method.invoke(Native Method)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
> 03-16 00:17:02.916 21430 21430 D NearbyParentFragment: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
> ```
> 
> Maybe database migration code is needed?

Try to login again, then it will work fine

File: app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java (hunk 4 above)
Comment (Pratham2305) [reply app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java]: Added!

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Otherwise it seems to run great, the database migration from master works correctly, the items with a P582 appear in the right color.

I have not run the unit tests, though.

**Review by nicolas-raoul — APPROVED** [review]
Unit tests run fine too.

**Comment by nicolas-raoul** [comment]
Merging as I believe that Neslihan's requested changes have been performed too.
Thanks Pratham and sorry for the time it took us to perform the review!

**Pull request merged**